webapp/src/ts/services/fast-action-button.service.ts

```diff
@@ -0,0 +1,236 @@
+import { Inject, Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { DOCUMENT } from '@angular/common';
+import { Store } from '@ngrx/store';
+
+import { AuthService } from '@mm-services/auth.service';
+import { ResponsiveService } from '@mm-services/responsive.service';
+import { ReportsActions } from '@mm-actions/reports';
+import { ButtonType } from '@mm-components/fast-action-button/fast-action-button.component';
+import { TranslateService } from '@mm-services/translate.service';
+import { TranslateFromService } from '@mm-services/translate-from.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FastActionButtonService {
+
+  private reportsActions: ReportsActions;
+
+  constructor(
+    private router: Router,
+    private store: Store,
+    private authService: AuthService,
+    private responsiveService: ResponsiveService,
+    private translateService: TranslateService,
+    private translateFromService: TranslateFromService,
+    @Inject(DOCUMENT) private document: Document,
+  ) {
+    this.reportsActions = new ReportsActions(store);
+  }
+
+  private executeMailto(link) {
+    const element = this.document.createElement('a');
+    element.href = link;
+    element.click();
+    element.remove();
+  }
+
+  private async filterActions(actions: FastAction[]): Promise<FastAction[]> {
+    const filteredActions = [];
+
+    for (const action of actions) {
+      if (await action.canDisplay()) {
+        filteredActions.push(action);
+      }
+    }
+
+    return filteredActions;
+  }
+
+  private getFormTitle(labelKey?: string, label?: string): string|undefined {
+    if (labelKey) {
+      return this.translateService.instant(labelKey);
+    }
+
+    if (label) {
+      return this.translateFromService.get(label);
+    }
+  }
+
+  private getReportFormActions(xmlForms, callbackContactReportModal?): FastAction[] {
+    return (xmlForms || [])
```

Comment:
```suggestion
  private getReportFormActions(xmlForms = [], callbackContactReportModal?): FastAction[] {
    return xmlForms
```

Reply:
I tried this, but the lint is complaining that default values should go last, but the other parameter is optional.
So, in the end, I had to add default values to both parameters `private getReportFormActions(xmlForms = [], callbackContactReportModal = null)`